dals/fileFunctions.js

```diff
@@ -1,36 +1,66 @@
 const mongoose = require('mongoose');
-const Issue = mongoose.model('Issue');
-const File = mongoose.model('File');
-
-exports.saveFile = (req, res) => {
-    const data = req.payload;
-    if(data.file){
-        File.save({
-            url: data.file.hapi.filename,
-            issue: issue._id
-        }, (err, issue) => {
-            if(err){
-                reply(err).code(500);
-            }
-            return res.response(issue);
+const fs = require('fs');
+const mime = require('mime-types');
+
+const path = require('path');
+const Issue = require(path.resolve('./models/issueModel'));
+const File = require('../models/fileModel');
+
+const internals = {};
+
+internals.saveFile = (file, issueId, resolve, fileName) => {
+    File.create({
+        path: fileName,
+        issue: issueId
+    }, (err, file) => {
+        if(err){
+            return reject(err);
+        }
+
+        Issue.findOne({ _id: issueId })
+        .then((issue) => {
+            issue.files.push(file._id);
+            issue.save().then((data) => {
+                return resolve(file);
+            });
         });
-    }
-}
+    });
+};
 
 exports.uploadFile = (req, res) => {
```

Comment:
Dals functions shouldn't accept req, res as parameters since these are high level arguments. They are handled in controller. In conroller you should extract things you need from req and pass it lower down to facade... Something like this:

```
// some controller
const someParamsAlpha = req.params.someParamsAlpha;
const someParamsBeta = req.params.someParamsBeta;

someFacade.someFunction(someParamsAlpha, someParamsBeta);
// then facade can call dals with these parameter
```

Makes sense?